src/mon/Monitor.cc

```diff
@@ -3697,6 +3700,59 @@ void Monitor::handle_command(MonOpRequestRef op)
     rdata.append(ds);
     rs = "";
     r = 0;
+  } else if (prefix == "mon ok-to-stop") {
```

Comment:
but i still think that we should warn user *and* return `-EINVAL` 

* if one or more invalid monitors not in quorum are passed to "ok-to-stop".
* if a monitor not in monmap is passed to "ok-to-rm".

Reply:
> but i still think that we should warn user _and_ return `-EINVAL`
> 
> * if one or more invalid monitors not in quorum are passed to "ok-to-stop".

The reason I left it off is it's possible to have a, say, mon.d started and running but not part of the monmap, e.g., because it is new and still synchronizing and hasn't yet added itself to the monmap.  In that case, it's safe to stop.  (I'm imagining these gates being used by rook to check whether it can restart/reschedule pods.)

> * if a monitor not in monmap is passed to "ok-to-rm".

Same reason as above.  Also, it means a consumer doing the ok-to-rm check and then rm doesn't have to to check whether the thing to remove was already removed.  We don't normally require that sequences of commands are idempotent, but in this case it would simplify the caller.